src/scripts/public/single.js

```diff
@@ -0,0 +1,41 @@
+import { qs, addSafeEventListener } from './utils';
+
+/**
+ * Single post manipulation.
+ * [@return](https://github.com/return) {void}
+ */
+const bcgovBlockThemeSinglePost = () => {
+    /*
+     * SafarIE iOS requires window.requestAnimationFrame update.
+     */
+    window.requestAnimationFrame( () => {
+        /**
+         * Check if this is a single post.
+         */
+        const body = qs( '.single-post' );
```

Comment:
not sure i understand why we are adding a js file for css, we shouldn't have to do this.

Reply:
We need to grab the header height and subtract that from the document viewport height. That can't be done with pure CSS.

Reply:
I analyzed the structure of the repo and found several instances of this. Check out dom-loader.js for example. I structured my code to match the other js files as well.

Reply:
@codewisenate is there a reason we would have to do this for every type of page, would you not already be doing this in a generic sense,  if in fact you really need to do this.

Reply:
This is not bad practice if you want you view height to to be specific across sites that use the same codebase and may have variable or dynamic sizing of in page components. As the block theme does not prescribe exact dimensions on the header it is not possible to simply add a css method like calc() with preset values. The CSS can offer a fallback, but offering exact dimensions by way of a JS calculation to augment that with page load details is not taxing the site and offers a fine tuned value for general use.